private/waiter/waiter.go

```diff
@@ -61,20 +61,56 @@ func (w *Waiter) Wait() error {
 					break
 				}
 				result = true
-				for _, val := range vals {
-					if !awsutil.DeepEqual(val, a.Expected) {
-						result = false
-						break
+				if exp, ok := a.Expected.(string); ok {
```

Comment:
Under which condition do you see this happening? I've searched the SDK and all I could find was that the waiter waits for either a string or an int. I'm rather tempted to split this into two fields actually, ExpectString and ExpectInt.